src/backend/app/auth/roles.py

```diff
@@ -334,3 +334,60 @@ async def mapper(
         current_user,
         ProjectRole.MAPPER,
     )
+
+
+async def project_contributors(
+    project_user: Annotated[ProjectUserDict, Depends(mapper)],
+    db: Annotated[Connection, Depends(db_conn)],
+    current_user: Annotated[AuthUser, Depends(login_required)],
+) -> bool:
+    """A contributor to a specific project."""
+    user_id = current_user.id
+    project = project_user["project"]
+    project_id = project.id
+    org_id = project.organisation_id
+
+    query = """
+        SELECT
+            EXISTS (
+                SELECT 1 FROM projects
+                WHERE author_id = %(user_id)s
+            ) OR
+            EXISTS (
+                SELECT 1 FROM users
+                WHERE id = %(user_id)s
+                  AND role = 'ADMIN'
+            ) OR
+            EXISTS (
+                SELECT 1 FROM user_roles
+                WHERE user_id = %(user_id)s
+                  AND project_id = %(project_id)s
+                  AND role = 'PROJECT_MANAGER'
+            ) OR
+            EXISTS (
+                SELECT 1 FROM organisation_managers
+                WHERE user_id = %(user_id)s
+                  AND organisation_id = %(org_id)s
+            ) OR
+            EXISTS (
+                SELECT 1 FROM task_events
+                WHERE user_id = %(user_id)s
+                  AND project_id = %(project_id)s
+            ) AS has_permission;
+    """
+    async with db.cursor() as cur:
+        await cur.execute(
+            query,
+            {"user_id": user_id, "project_id": project_id, "org_id": org_id},
+        )
+        result = await cur.fetchone()
+
+    print("result is")
+    print(result, result[0])
+    if result and result[0]:
+        return True
```

Comment:
instead of returning True we can return user_dict and use that to get information of user, project.